src/overlays/actors/ovl_Demo_Getitem/z_demo_getitem.c

```diff
@@ -27,16 +30,84 @@ const ActorInit Demo_Getitem_InitVars = {
 };
 
-#endif
+static s16 sObjectBankIndices[] = { OBJECT_GI_MASK14, OBJECT_GI_SWORD_4 };
+
+static s16 sGetItemDraws[] = { GID_MASK_GREAT_FAIRY, GID_SWORD_GREAT_FAIRY };
+
+static u16 csActionIndex[] = { 0x6E, 0x236 };
+
+void DemoGetitem_Init(Actor* thisx, PlayState* play) {
+    s32 pad;
+    s32 objIndex;
+    s32 itemIndex;
+    DemoGetitem* this = THIS;
+
+    itemIndex = 0;
+    if (DEMOGETITEM_GET_F(thisx) == 1) {
+        itemIndex = 1;
+    }
+    Actor_SetScale(&this->actor, 0.25f);
+    this->actionFunc = func_80A4FB10;
+    this->item = sGetItemDraws[itemIndex];
+    this->csAction = csActionIndex[itemIndex];
+    objIndex = Object_GetIndex(&play->objectCtx, sObjectBankIndices[itemIndex]);
+    if (objIndex < 0) {
+        Actor_MarkForDeath(&this->actor);
+    } else {
+        this->object = objIndex;
+    }
+}
+
+void DemoGetitem_Destroy(Actor* thisx, PlayState* play) {
+}
+
+void func_80A4FB10(DemoGetitem* this, PlayState* play) {
+    if (Object_IsLoaded(&play->objectCtx, this->object)) {
+        this->actor.draw = NULL;
+        this->actor.objBankIndex = this->object;
+        this->actionFunc = func_80A4FB68;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Demo_Getitem/DemoGetitem_Init.s")
+void func_80A4FB68(DemoGetitem* this, PlayState* play2) {
+    PlayState* play = play2;
+    u16 sp22 = (play->gameplayFrames * 1000) & 0xFFFF;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Demo_Getitem/DemoGetitem_Destroy.s")
+    if (Cutscene_CheckActorAction(play, this->csAction)) {
+        if (play->csCtx.actorActions[Cutscene_GetActorActionIndex(play, this->csAction)]->action != 4) {
+            this->actor.shape.yOffset = 0.0f;
+        }
+        switch (play->csCtx.actorActions[Cutscene_GetActorActionIndex(play, this->csAction)]->action) {
+            case 2:
+                this->actor.draw = func_80A4FCF0;
+                Cutscene_ActorTranslate(&this->actor, play, Cutscene_GetActorActionIndex(play, this->csAction));
+                this->actor.shape.rot.y += 1000;
+                return;
+            case 3:
+                Actor_MarkForDeath(&this->actor);
+                return;
+            case 4:
+                this->actor.draw = func_80A4FCF0;
+                Cutscene_ActorTranslateAndYaw(&this->actor, play, Cutscene_GetActorActionIndex(play, this->csAction));
+                this->actor.shape.yOffset = Math_SinS(sp22) * 15.0f;
+                return;
+        }
+        this->actor.draw = NULL;
+    } else {
+        this->actor.draw = NULL;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Demo_Getitem/func_80A4FB10.s")
+void DemoGetitem_Update(Actor* thisx, PlayState* play) {
+    DemoGetitem* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Demo_Getitem/func_80A4FB68.s")
+    this->actionFunc(this, play);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Demo_Getitem/DemoGetitem_Update.s")
+void func_80A4FCF0(Actor* thisx, PlayState* play) {
```

Comment:
This may not be set in the initvars, but it's the only draw function, so `DemoGetitem_Draw`.